testutil/daemon/daemon.go

```diff
@@ -534,6 +530,15 @@ func (d *Daemon) StartWithLogFile(out *os.File, providedArgs ...string) error {
 		d.args = append(d.args, "--storage-driver", d.storageDriver)
 	}
 
+	// If no --firewall-backend is configured and env var DOCKER_FIREWALL_BACKEND has a value, use it.
+	if !slices.ContainsFunc(providedArgs, func(s string) bool {
+		return strings.HasPrefix(s, "--firewall-backend")
+	}) {
```

Comment:
nit: Instead of a comment, this could be clearer with an intermediate variable.

```suggestion
    hasFwBackendArg := slices.ContainsFunc(providedArgs, func(s string) bool {
        return strings.HasPrefix(s, "--firewall-backend")
	})
	if !hasFwBackendArg {
```

(just a subjective nit though, @thaJeztah would probably disagree 😅)

Reply:
The only opinion wins! I've changed it.